code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -12,15 +12,234 @@ GLOBAL_LIST_INIT(blacklisted_borg_hats, typecacheof(list( //Hats that don't real
+	if(W.tool_behaviour == TOOL_WELDER && (user.a_intent != INTENT_HARM || user == src))
+		user.changeNext_move(CLICK_CD_MELEE)
+		if (!getBruteLoss())
+			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
+			return
+		if (!W.tool_start_check(user, amount=0)) //The welder has 1u of fuel consumed by it's afterattack, so we don't need to worry about taking any away.
+			return
+		if(src == user)
+			to_chat(user, "<span class='notice'>You start fixing yourself...</span>")
+			if(!W.use_tool(src, user, 50))
+				return
+
+		adjustBruteLoss(-30)
+		updatehealth()
+		add_fingerprint(user)
+		visible_message("<span class='notice'>[user] has fixed some of the dents on [src].</span>")
+		return
+
+	if(istype(W, /obj/item/stack/cable_coil) && wiresexposed)
+		user.changeNext_move(CLICK_CD_MELEE)
+		var/obj/item/stack/cable_coil/coil = W
+		if (getFireLoss() > 0 || getToxLoss() > 0)
+			if(src == user)
+				to_chat(user, "<span class='notice'>You start fixing yourself...</span>")
+				if(!do_after(user, 50, target = src))
+					return
+			if (coil.use(1))
+				adjustFireLoss(-30)
+				adjustToxLoss(-30)
+				updatehealth()
+				user.visible_message("<span class='notice'>[user] has fixed some of the burnt wires on [src].</span>", "<span class='notice'>You fix some of the burnt wires on [src].</span>")
+			else
+				to_chat(user, "<span class='warning'>You need more cable to repair [src]!</span>")
+		else
+			to_chat(user, "<span class='warning'>The wires seem fine, there's no need to fix them.</span>")
+		return
+
+	if(W.tool_behaviour == TOOL_CROWBAR)	// crowbar means open or close the cover
+		if(opened)
+			to_chat(user, "<span class='notice'>You close the cover.</span>")
+			opened = 0
+			update_icons()
+		else
+			if(locked)
+				to_chat(user, "<span class='warning'>The cover is locked and cannot be opened!</span>")
+			else
+				to_chat(user, "<span class='notice'>You open the cover.</span>")
+				opened = 1
+				update_icons()
+		return
+
+	if(istype(W, /obj/item/stock_parts/cell) && opened)	// trying to put a cell inside
+		if(wiresexposed)
+			to_chat(user, "<span class='warning'>Close the cover first!</span>")
+		else if(cell)
+			to_chat(user, "<span class='warning'>There is a power cell already installed!</span>")
+		else
+			if(!user.transferItemToLoc(W, src))
+				return
+			cell = W
+			to_chat(user, "<span class='notice'>You insert the power cell.</span>")
+		update_icons()
+		diag_hud_set_borgcell()
+		return
+
+	if(is_wire_tool(W))
+		if (wiresexposed)
+			wires.interact(user)
+		else
+			to_chat(user, "<span class='warning'>You can't reach the wiring!</span>")
+		return
+
+	if(W.tool_behaviour == TOOL_SCREWDRIVER && opened && !cell)	// haxing
+		wiresexposed = !wiresexposed
+		to_chat(user, "<span class='notice'>The wires have been [wiresexposed ? "exposed" : "unexposed"].</span>")
+		update_icons()
+		return
+
+	if(W.tool_behaviour == TOOL_SCREWDRIVER && opened && cell)	// radio
```

Comment:
oof

Reply:
Yeah, that's a bit strange.

To be clear, most of this isn't my code. We just had two identical procs in different files, and I figured I'd merge the two since I was touching one of them anyway. The only change I made was removing all the `else if`s.

Reply:
dupe checks will still need to be fixed

Reply:
Alright, fixed